### PR TITLE
[10.0] [ADD] mrp_operating_unit

### PR DESCRIPTION
**Pull request by nikul-serpentcs**
MRP with Operating Units
========================

This module introduces the following features:

* Adds Operating Unit (OU) to the Manufacturing Order created from the procurement. 

* This module implements global security rules on manufacturing orders so that a user can only read manufacturing orders where the location is linked to an
  operating unit that the user has access to.

### CODE REVIEW

**Comment by nikul-serpentcs** [comment]
@aheficent Could you please review.

File: mrp_operating_unit/README.rst [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+
+* Adds Operating Unit (OU) to the Manufacturing Order created from the procurement. 
+
+* This module implements global security rules on manufacturing orders so that a user can only read manufacturing orders where the location is linked to an
+  operating unit that the user has access to.
+
+Installation
+============
+
+No specific installation requirements.
```
Comment (AaronHForgeFlow) [inline mrp_operating_unit/README.rst]: Please remove empty sections

**Review by AaronHForgeFlow — APPROVED** [review]
LGTM :+1: Codecov fails because this module is new

**Review by AaronHForgeFlow — CHANGES_REQUESTED** [review]
Please add your changes after the commits here https://github.com/OCA/operating-unit/pull/39

File: mrp_operating_unit/models/mrp.py [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+from odoo import api, fields, models
+from odoo.tools.translate import _
+from odoo.exceptions import ValidationError
+
+
+class MrpProduction(models.Model):
+
+    _inherit = 'mrp.production'
+
+    operating_unit_id = fields.Many2one('operating.unit', 'Operating Unit',
```
Comment (LoisRForgeFlow) [inline mrp_operating_unit/models/mrp.py]: I would make this field readonly once the MO is confirmed

File: mrp_operating_unit/models/mrp.py [hunk 3]
```diff
@@ -11,10 +11,13 @@ class MrpProduction(models.Model):
...
     _inherit = 'mrp.production'
 
-    operating_unit_id = fields.Many2one('operating.unit', 'Operating Unit',
-                                        default=lambda self:
-                                        self.env['res.users'].
-                                        operating_unit_default_get(self._uid))
+    operating_unit_id = fields.Many2one(
+        'operating.unit', 'Operating Unit',
+        readonly=True,
+        state={'confirmed': [('readonly', True)]},
```
Comment (LoisRForgeFlow) [inline mrp_operating_unit/models/mrp.py]: `states={'draft': [('readonly', False)]}`

**Comment by AaronHForgeFlow** [comment]
@nikul-serpentcs please fix the commit history so we can approve this :smiley:

**Comment by AaronHForgeFlow** [comment]
@nikul-serpentcs  No need to close this you can fix this branch

**Comment by AaronHForgeFlow** [comment]
Nicely done! :medal_sports:

**Comment by nikul-serpentcs** [comment]
Thanks for the medal!

> @nikul-serpentcs No need to close this you can fix this branch

you didn't see that coming ;)

File: mrp_operating_unit/__manifest__.py [hunk 4]
```diff
@@ -1,20 +1,21 @@
 # -*- coding: utf-8 -*-
-# © 2015 Eficent Business and IT Consulting Services S.L. -
-# Jordi Ballester Alomar
-# © 2015 Serpent Consulting Services Pvt. Ltd. - Sudhir Arya
+# Copyright 2017 Eficent Business and IT Consulting Services S.L.
+# Copyright 2017 Serpent Consulting Services Pvt. Ltd.
```
Comment (LoisRForgeFlow) [inline mrp_operating_unit/__manifest__.py]: Copyright started in 2015 so: `Copyright 2015-17 (...)` in all files

@nikul-serpentcs

Comment (nikul-serpentcs) [reply mrp_operating_unit/__manifest__.py]: @lreficent Please review Copyright.

**Review by LoisRForgeFlow — APPROVED** [review]
OK, let's move forward with this. Thanks for attending all comments @nikul-serpentcs

**Review by AaronHForgeFlow — APPROVED** [review]
:+1:

**Review by MiquelRForgeFlow — APPROVED** [review]
LGTM

**Pull request merged**